assets/src/components/with-animation-controls.js

```diff
@@ -1,22 +1,91 @@
 /**
  * WordPress dependencies
  */
-import { createHigherOrderComponent } from '@wordpress/compose';
+import { createHigherOrderComponent, compose } from '@wordpress/compose';
 import { InspectorControls } from '@wordpress/editor';
 import { Fragment } from '@wordpress/element';
 import { PanelBody } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { withDispatch, withSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { ALLOWED_CHILD_BLOCKS } from '../constants';
-import { AnimationControls, withParentBlock } from './';
+import { AnimationControls } from './';
+
+const applyWithSelect = withSelect( ( select, props ) => {
+	const { getSelectedBlockClientId, getBlockRootClientId, getBlock, getBlocksByClientId, getBlockOrder } = select( 'core/editor' );
+	const { getAnimationOrder } = select( 'amp/story' );
+
+	const currentBlock = getSelectedBlockClientId();
+	const page = getBlockRootClientId( currentBlock );
+
+	const { ampAnimationAfter } = props.attributes;
+	const predecessor = getBlocksByClientId( getBlockOrder( page ) ).find( ( b ) => b.attributes.anchor === ampAnimationAfter );
```

Comment:
What is `b`?

Reply:
b as in block. I think I shortened it originally because the variable block was already being used in the outer scope.